### PR TITLE
Firecracker: add basic tracing

### PR DESCRIPTION
**Pull request by bduffany**
Tracing is very basic at the moment to give us a rough idea of where all the time is being spent.

Initial trace results show that a good chunk of time is spent setting up networking and connecting to the VM exec server (which may partly be due to init time), so we might want to spend some time optimizing those.

Example trace:

image

---

**Version bump**: Patch

### CODE REVIEW

File: enterprise/server/remote_execution/containers/firecracker/firecracker.go [hunk 1]
```diff
@@ -1075,29 +1107,38 @@ func (c *FirecrackerContainer) Create(ctx context.Context, actionWorkingDir stri
...
 	}
-	if err := m.Start(vmCtx); err != nil {
+	if err := c.startMachine(ctx, vmCtx, m); err != nil {
 		return status.InternalErrorf("Failed starting machine: %s", err)
 	}
 	c.machine = m
 	return nil
 }
 
+func (c *FirecrackerContainer) startMachine(ctx context.Context, startCtx context.Context, m *fcclient.Machine) error {
```
Comment (tylerwilliams) [inline enterprise/server/remote_execution/containers/firecracker/firecracker.go:1117]: might be clearer to not put this in a separate method like this. The two contexts coming from somewhere else are a bit mind boggling.

(also, not relevant if you remove this, but when you have more than one sequential arguments to a function in go that have the same type, you can omit the type information for all but the last one to save space)

Comment (bduffany) [reply enterprise/server/remote_execution/containers/firecracker/firecracker.go]: Moved the tracing back to the original callsite, with the help of a util func that wraps an anonymous func with a span. wdyt about this approach?

Comment (bduffany) [reply enterprise/server/remote_execution/containers/firecracker/firecracker.go]: Resolved offline, using an inline closure instead

**Review by tylerwilliams — APPROVED** [review]
LGTM

(this is the way)

**Pull request merged**